src/visualize_paths.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+//jscs:disable requireSpacesInForStatement
+//jscs:disable requireSpaceBeforeBinaryOperators
+//jscs:disable disallowMultipleVarDecl
+//jscs:disable validateQuoteMarks
+//jscs:disable requireSpaceBeforeObjectValues
+//jscs:disable validateIndentation
+//jscs:disable requireCurlyBraces
+//jscs:disable requireSpaceAfterKeywords
+//jscs:disable requireSpaceBetweenArguments
+//jscs:disable requireSpacesInConditionalExpression
+//jscs:disable requireSpaceAfterBinaryOperators
+//jscs:disable disallowSpaceAfterPrefixUnaryOperators
+//jscs:disable validateParameterSeparator
+//jscs:disable requireSpaceBeforeBlockStatements
+//jscs:disable disallowSpacesInsideObjectBrackets
+//jscs:disable disallowSpacesInAnonymousFunctionExpression
+//jscs:disable disallowMultipleLineBreaks
+//jscs:disable disallowSpacesInFunctionExpression
+global.visualizePaths = function () {
+  let Visual = require('visual');
+  let colors = [];
+  let COLOR_BLACK = colors.push('#000000') - 1;
+  let COLOR_RED = colors.push('rgba(249,8,8,0.5)') - 1;
+  let COLOR_PATH = colors.push('rgba(255,255,255,0.5)') - 1;
+  _.each(Game.rooms, (room, name) => {
+    let visual = new Visual(name);
+    visual.defineColors(colors);
+    visual.setLineWidth = 0.5;
+    _.each(Game.creeps, creep=> {
+      if (creep.room != room) return;
+      let mem = creep.memory;
+      if (mem._move) {
+        let path = Room.deserializePath(mem._move.path);
+        if (path.length) {
+          //console.log('drawing route ' + mem._move.path);
+          visual.drawLine(path.map(p=>([p.x, p.y])), COLOR_RED, { lineWidth: 0.1 });
+        }
+      }
+    });
+    if (room.memory.routing)
+      for (var route in room.memory.routing) {

```

Comment:
Just as an idea, you could use: https://github.com/TooAngel/screeps/blob/master/src/prototype_room_memory.js

This would use cached paths additional to the fixed ones and manages the deserialization for you.